org.osgi.util.converter/src/org/osgi/util/converter/ConvertingImpl.java

```diff
@@ -827,30 +827,40 @@ public Object invoke(Object proxy, Method method, Object[] args)
 					if (cls.isAnnotation()) {
 						val = method.getDefaultValue();
 					} else if (method.isDefault()) {
-						double javaVersion = Double.parseDouble(
-								System.getProperty("java.class.version"));
-						double java8 = 52.0d;
-						if (javaVersion > java8) {
-							val = MethodHandles.lookup()
-									.findSpecial(method.getDeclaringClass(),
-											method.getName(),
-											MethodType.methodType(
-													method.getReturnType(),
-													new Class[] {}),
-											method.getDeclaringClass())
-									.bindTo(proxy)
-									.invokeWithArguments(args);
-						} else {
-							Constructor<Lookup> constructor = Lookup.class
-									.getDeclaredConstructor(Class.class);
-							if (!constructor.isAccessible()) {
-								constructor.setAccessible(true);
+						try {
+							double javaVersion = Double.parseDouble(
+									System.getProperty("java.class.version"));
+							double java8 = 52.0d;
+							if (javaVersion > java8) {
+								val = MethodHandles.lookup()
+										.findSpecial(method.getDeclaringClass(),
+												method.getName(),
+												MethodType.methodType(
+														method.getReturnType(),
+														new Class[] {}),
+												method.getDeclaringClass())
+										.bindTo(proxy)
+										.invokeWithArguments(args);
+							} else {
+								Constructor<Lookup> constructor = Lookup.class
+										.getDeclaredConstructor(Class.class);
+								if (!constructor.isAccessible()) {
+									constructor.setAccessible(true);
+								}
+								val = constructor.newInstance(cls)
+										.in(cls)
+										.unreflectSpecial(method, cls)
+										.bindTo(proxy)
+										.invokeWithArguments(args);
 							}
-							val = constructor.newInstance(cls)
-									.in(cls)
-									.unreflectSpecial(method, cls)
-									.bindTo(proxy)
-									.invokeWithArguments(args);
+							if (val == null) {
+								return val;
```

Comment:
I am not sure this is even correct. A null value can be converted into other things. For conversion to boolean, null is converted to false. For conversion to a collection, null is converted to an empty collection. So I think you must always fall through to the final conversion at the end of this method.